core/azure-core/src/main/java/com/azure/core/implementation/configuration/ConfigurationManager.java

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.configuration;
+
+import com.azure.core.implementation.logging.ServiceLogger;
+import com.azure.core.implementation.util.ImplUtils;
+
+import java.util.function.Function;
+
+/**
+ * Manages the global configuration store.
+ */
+public final class ConfigurationManager {
+    private static final ServiceLogger LOGGER = new ServiceLogger(ConfigurationManager.class);
+    private static Configuration configuration = new Configuration();
+
+    /**
+     * Gets the value of the configuration.
+     *
+     * @param name Name of the configuration.
+     * @return Value of the configuration if found, otherwise null.
+     */
+    public static String get(String name) {
+        return getOrLoad(name);
+    }
+
+    /**
+     * Gets the value of the configuration converted to {@code T}.
+     *
+     * If no configuration is found the default is returned.
+     *
+     * @param name Name of the configuration.
+     * @param defaultValue Value to return if the configuration isn't found.
+     * @param <T> Generic type that the configuration is converted to if found.
+     * @return The converted configuration if found, otherwise the default value is returned.
+     */
+    public static <T> T get(String name, T defaultValue) {
+        return Configuration.convertOrDefault(getOrLoad(name), defaultValue);
+    }
+
+    /**
+     * Gets the converted value of the configuration.
+     *
+     * @param name Name of the configuration.
+     * @param converter Converter used to map the configuration to {@code T}.
+     * @param <T> Generic type that the configuration is converted to if found.
+     * @return The converted configuration if found, otherwise null.
+     */
+    public static <T> T get(String name, Function<String, T> converter) {
+        return converter.apply(getOrLoad(name));
+    }
+
+    /**
+     * @return a mutable clone of the global configuration store.
+     */
+    public static Configuration configuration() {
```

Comment:
This should be called `getConfiguration()` or `getGlobalConfiguration()` or similar.

Reply:
Changed to getConfiguration, I also changed it to return the Configuration instance as well instead of a clone so the Configuration is able to be mutated.